### ISSUE TITLE
Undefined symbol in libtensorflow_ops.so for gpu versions 0.3.0 and 0.4.0

### ISSUE THREAD

**Issue by Spiess** [eaplatanios/tensorflow_scala#137]
When using any TensorFlow features on TensorFlow Scala GPU versions 0.3.0 and 0.4.0 I encounter the following error:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.platanios.tensorflow.jni.Graph$.<init>(Graph.scala:22)
	at org.platanios.tensorflow.jni.Graph$.<clinit>(Graph.scala)
	at org.platanios.tensorflow.api.core.Graph$.apply(Graph.scala:902)
	at org.platanios.tensorflow.api.core.package$.<init>(package.scala:26)
	at org.platanios.tensorflow.api.core.package$.<clinit>(package.scala)
	at org.platanios.tensorflow.api.ops.package$.<init>(package.scala:44)
	at org.platanios.tensorflow.api.ops.package$.<clinit>(package.scala)
	at org.platanios.tensorflow.api.ops.Op$Builder.<init>(Op.scala:2574)
	at org.platanios.tensorflow.api.ops.basic.Constructors.placeholder(Constructors.scala:396)
	at org.platanios.tensorflow.api.ops.basic.Constructors.placeholder$(Constructors.scala:392)
	at org.platanios.tensorflow.api.package$tf$.placeholder(package.scala:293)
	at test.LinearRegression$.main(LinearRegression.scala:36)
	at test.LinearRegression.main(LinearRegression.scala)
Caused by: org.platanios.tensorflow.jni.NotFoundException: /tmp/tensorflow_scala_native_libraries6163828859171220367/libtensorflow_ops.so: undefined symbol: _ZN10tensorflow12OpDefBuilder4AttrESs
	at org.platanios.tensorflow.jni.TensorFlow$.loadOpLibrary(Native Method)
	at org.platanios.tensorflow.jni.TensorFlow$.$anonfun$load$6(TensorFlow.scala:107)
	at scala.Option.foreach(Option.scala:257)
	at org.platanios.tensorflow.jni.TensorFlow$.load(TensorFlow.scala:107)
	at org.platanios.tensorflow.jni.TensorFlow$.<init>(TensorFlow.scala:155)
	at org.platanios.tensorflow.jni.TensorFlow$.<clinit>(TensorFlow.scala)
	... 13 more
```

This only happens when using the GPU version of 0.3.0 and 0.4.0. All CPU versions as well as the GPU version of 0.2.4 are working fine. The newest Python GPU TensorFlow is also running without problems.

I am using x86_64 linux with CUDA 9.0 and NVIDIA driver version 410.73.

**Comment by eaplatanios** [comment]
@Spiess This is unfortunately due to an issue with the official nightly precompiled binaries distributed by Google. There’s been an ongoing discussion to standardize the process for publishing them and so the issue should be fixed soon in 0.4.1. I’m currently traveling and can not provide a better temporary solution, but for now you could try compiling TF from source, from commit 4b46a399d7a5a5c9cc262d08e0df1217364a5b9f . That should work. :)

**Comment by eaplatanios** [comment]
This should have been fixed in the latest release (0.4.2) and so I'll close this for now. Please feel free to reopen if the issue persists. Note that currently, only Scala 2.12 binaries have been released (and not 2.11), due to some issues with implicit derivation in Scala 2.11.
